pkg/intermediate/aggregate.go

```diff
@@ -0,0 +1,284 @@
+	if containsInfoElement(infoElementMap, "sourceIPv4Address") && containsInfoElement(infoElementMap, "destinationIPv4Address") && containsInfoElement(infoElementMap, "sourceTransportPort") && containsInfoElement(infoElementMap, "destinationTransportPort") && containsInfoElement(infoElementMap, "protocolIdentifier") {
+		srcIPAddr, ok := infoElementMap["sourceIPv4Address"].Value.(net.IP)
+		if !ok {
+			return Tuple{}, fmt.Errorf("sourceIPv4Address is not in correct format.")
+		}
+		srcIP = srcIPAddr.String()
+		dstIPAddr, ok := infoElementMap["destinationIPv4Address"].Value.(net.IP)
+		if !ok {
+			return Tuple{}, fmt.Errorf("destinationIPv4Address is not in correct format.")
+		}
+		dstIP = dstIPAddr.String()
+		srcPortNum, ok := infoElementMap["sourceTransportPort"].Value.(uint16)
+		if !ok {
+			return Tuple{}, fmt.Errorf("sourceTransportPort is not in correct format.")
+		}
+		srcPort = srcPortNum
+		dstPortNum, ok := infoElementMap["destinationTransportPort"].Value.(uint16)
+		if !ok {
+			return Tuple{}, fmt.Errorf("destinationTransportPort is not in correct format.")
+		}
+		dstPort = dstPortNum
+		protoNum, ok := infoElementMap["protocolIdentifier"].Value.(uint8)
+		if !ok {
+			return Tuple{}, fmt.Errorf("protocolIdentifier is not in correct format.")
+		}
+		proto = protoNum
+	} else if containsInfoElement(infoElementMap, "sourceIPv6Address") && containsInfoElement(infoElementMap, "destinationIPv6Address") && containsInfoElement(infoElementMap, "sourceTransportPort") && containsInfoElement(infoElementMap, "destinationTransportPort") && containsInfoElement(infoElementMap, "protocolIdentifier") {
+		srcIPAddr, ok := infoElementMap["sourceIPv6Address"].Value.(net.IP)
+		if !ok {
+			return Tuple{}, fmt.Errorf("sourceIPv6Address is not in correct format.")
+		}
+		srcIP = net.IP(srcIPAddr).String()
+		dstIPAddr, ok := infoElementMap["destinationIPv6Address"].Value.(net.IP)
+		if !ok {
+			return Tuple{}, fmt.Errorf("destinationIPv6Address is not in correct format.")
+		}
+		dstIP = net.IP(dstIPAddr).String()
+		srcPortNum, ok := infoElementMap["sourceTransportPort"].Value.(uint16)
+		if !ok {
+			return Tuple{}, fmt.Errorf("sourceTransportPort is not in correct format.")
+		}
+		srcPort = srcPortNum
+		dstPortNum, ok := infoElementMap["destinationTransportPort"].Value.(uint16)
+		if !ok {
+			return Tuple{}, fmt.Errorf("destinationTransportPort is not in correct format.")
+		}
+		dstPort = dstPortNum
+		protoNum, ok := infoElementMap["protocolIdentifier"].Value.(uint8)
+		if !ok {
+			return Tuple{}, fmt.Errorf("protocolIdentifier is not in correct format.")
+		}
+		proto = protoNum
+	} else {
+		return Tuple{}, fmt.Errorf("missing 5-tuple value(s) in the record.")
+	}
+	return Tuple{srcIP, dstIP, proto, srcPort, dstPort}, nil
+}
+
+// addOriginalExporterInfo adds originalExporterIPv4Address and originalObservationDomainId to records in message set
+func addOriginalExporterInfo(message *entities.Message) error {
+	set := message.Set
```

Comment:
Can a message has multiple Sets?

Reply:
Yes currently we only support one set per message. We should support multiple sets per message in the future. There is an issue to track it #55